src/cleanfigure.m

```diff
@@ -939,6 +939,30 @@ function removeNaNs(meta, handle)
     topRight    = [xLim(2); yLim(2)];
 end
 % =========================================================================
+function data = limitPrecision(data)
+  % Limit the precision of the given data
+  
+  % Only do something if the data is not empty
+  if isempty(data) || all(~isfinite(data))
+      return
```

Comment:
The only case that I saw where `max` returns `NaN` is if you do `max(NaN(1,N))`. In all other cases, `NaN` is omitted (see the help of `max`). But since there should never be a call to max which even has `NaN` or `Inf` to begin with, I don't see a case where that can happen.